src/bluefinClient.ts

```diff
@@ -1538,6 +1540,298 @@ export class BluefinClient {
     return this.contractCalls.withdrawAllFromMarginBankContractCall();
   };
 
+  closeAllDelistedPositionsAndWithdrawMargin = async (args?: {
+    gasBudget?: number;
+    sponsor?: boolean;
+    dryRunOnly?: boolean;
+  }) => {
+    try {
+      // get exchange info of all markets to read the status
+      const exchangeInfo = await this.getExchangeInfo();
+      const delistedMarkets = filterDelistedMarkets(exchangeInfo);
+
+      if (delistedMarkets.length <= 0) {
+        return {
+          ok: false,
+          code: 400,
+          data: "",
+          message: "No delisted markets",
+        };
+      }
+
+      //get user positions
+      const userPositions = await this.getUserPosition({});
+
+      let delistedUserPositionsSymbols: string[] = [];
+      //get delisted market positions
+      if (userPositions.data && userPositions.data.length > 0) {
+        delistedUserPositionsSymbols = userPositions.data
+          .filter((position) => {
+            return delistedMarkets.includes(position.symbol);
+          })
+          .map((position) => position.symbol);
+      }
+
+      if (delistedUserPositionsSymbols.length <= 0) {
+        return {
+          ok: false,
+          code: 400,
+          data: "",
+          message: "No delisted positions",
+        };
+      }
```

Comment:
I don't think we need to return here right? We can still go forward to cancel orders and withdraw any remaining margin if it exists right?

Reply:
Yep, I can call withdraw function here instead of our PTB

Reply:
same will be done in the other method, where I'll call `withdrawAllWusdcSwapAndDepositToPro` when there's no delisted position